File: coj/1016.py
from sys import stdin, stdout, maxint
from math import sqrt

def prim(vertex, n):
	mst = []
	mst_bool = [False for x in range(n)]
	mst_bool[0] = True
	mst.append(0)
	mst_distance = 0
	while(len(mst) != n):
		next_node, d = extract_min(vertex, mst, mst_bool, n)
		mst.append(next_node)
		mst_bool[next_node] = True
		mst_distance += d
	return mst_distance

def extract_min(vertex, mst, mst_bool, n):
	min_value, index = maxint, 0
	for node in mst:
		for i in range(n):
			if (node != i and not mst_bool[i]):
				d = distance(vertex[node], vertex[i])
				if (d < min_value):
					index = i
					min_value = d
	return index, min_value


def distance(p1, p2):
	return sqrt((p1[0]-p2[0])**2 + (p1[1]-p2[1])**2)

def main():
	n = int(stdin.readline().replace('\n',''))
	vertex = []
	for i in range(n):
		line = stdin.readline().replace('\n','')
		line = line.split(" ")
		vertex.append([float(line[0]),float(line[1])])
	mst_distance = prim(vertex, n)
	stdout.write('{0:.2f}\n'.format(mst_distance))

if __name__ == '__main__':
	main()

File: coj/1016.cpp
#include <iostream>
#include <cmath>
#include <cstring>
#include <limits>
#include <cstdlib>
#include <vector>
#include <iomanip>

using namespace std;
#define MAX 105

vector<pair<float,float> > graph(MAX);

float distance_two_points(pair<float, float> p1, pair<float, float> p2) {
	return sqrt((p1.first - p2.first) * (p1.first - p2.first) + (p1.second - p2.second) * (p1.second - p2.second));
}

void extract_min(bool mst[], int n, int* next_node, float* distance) {
	float min_value = numeric_limits<float>::max();
	int index;
	for(int i = 0; i < n; i++) {
		for(int j = 0; j < n; j++) {
			if (i != j && !mst[i]) {
				float d = distance_two_points(graph[i], graph[j]);
				if (d < min_value) {
					index = i;
					min_value = d;
				}
			}
		}
	}
	*next_node = index;
	*distance = min_value;
}

float prim(int n) {
	bool mst[n];
	float mst_distance = 0;
	memset(mst, false, sizeof(mst));
	int n_mst=1;
	mst[0] = true;
	while(n_mst != n) {
		int next_node;
		float distance;
		extract_min(mst, n, &next_node, &distance);
		mst[next_node] = true;
		mst_distance += distance;
		n_mst++;
	}
	return mst_distance;
}

int main() {
	int n;
	cin>>n;
	for(int i = 0; i < n; i++) {
		float x, y;
		cin>>x>>y;
		graph[i] = make_pair(x,y);
	}
	float distance = prim(n);
	cout<<fixed<<setprecision(2)<<distance<<endl;
	return 0;
}